source/language/pulses.rst

```diff
@@ -98,20 +98,18 @@ of prior versions of OpenQASM.
 Restrictions on defcal bodies
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The contents of ``defcal`` bodies are subject to some restrictions. This is
-similar to how boxed expressions work.
-
-- They must have a definite length, regardless of the parameters passed in or
-  the state of the system when called. This allows the compiler to properly
-  resolve ``lengthof(...)`` calls and allowing for optimizations.
-- No control flow is allowed within a ``defcal`` block (with one exception,
-  see below). This follows from the definite length requirement.
+The contents of ``defcal`` bodies are subject to the restriction they must have a definite length
+known at compile time, regardless of the parameters passed in or the state of the system when
+called. This allows the compiler to properly resolve ``lengthof(...)`` calls and
+allows for optimizations. If there is to be control flow in the ``defcal``, each branch of the
```

Comment:
I think we should revisit the question of control flow. You could likely do most of it outside of the `defcal` and leave the specific pulse sequence to be defined in the `defcal`. This would then maintain the prior requirement. I don't have a strong opinion either way though

Reply:
The main motivator for control flow inside the `defcal` is to be able to `defcal` the `reset` instruction.